_search-plugins/knn/approximate-knn.md

```diff
@@ -127,10 +127,14 @@ GET my-knn-index-1/_search
 }
 ```
 
-`k` is the number of neighbors the search of each graph will return. You must also include the `size` option, which
-indicates how many results the query actually returns. The plugin returns `k` amount of results for each shard
-(and each segment) and `size` amount of results for the entire query. The plugin supports a maximum `k` value of 10,000.
-Starting in OpenSearch 2.14, in addition to using the `k` variable, both the `min_score` and `max_distance` variables can be used for [radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/).
+In the preceding query, `k` is the number of neighbors the search of each graph returns. You must also include the `size` option, indicating the final number of results you want the query to return.  
+
+For NMSLIB and Faiss engines, `k` is the maximum number of documents returned for all segments of a shard. For the Lucene engine, `k` is the number of documents returned for a shard. The maximum value of `k` is 10,000.
```

Comment:
This line is not adding no new information if we don't add an example. An example like this:

| size | k  | Number of Shards | Number of segments per shard | Engine       | Final Number of Results Returned |
|------|----|------------------|------------------------------|--------------|----------------------------------|
| 10   | 1  | 1                | 4                            | Faiss/Nmslib | 4                                |
| 10   | 1  | 1                | 4                            | Lucene       | 1                                |
| 10   | 10 | 1                | 4                            | Faiss/Nmslib | 10                               |
| 10   | 10 | 1                | 4                            | Lucene       | 10                               |
| 10   | 1  | 2                | 4                            | Faiss/Nmslib | 8                                |
| 10   | 1  | 2                | 4                            | Lucene       | 2                                |

Reply:
@navneet1v Thank you for the additional information. I have added it to the docs. Please review again.